fl4health/clients/basic_client.py

```diff
@@ -536,69 +527,6 @@ def get_client_specific_reports(self) -> Dict[str, Any]:
         """
         return {}
 
-    def _move_data_to_device(
-        self, data: Union[TorchInputType, TorchTargetType]
-    ) -> Union[TorchTargetType, TorchInputType]:
-        """
-        Moving data to self.device where data is intended to be either input to
-        the model or the targets that the model is trying to achieve
-
-        Args:
-            data (TorchInputType | TorchTargetType): The data to move to
-                self.device. Can be a TorchInputType or a TorchTargetType
-
-        Raises:
-            TypeError: Raised if data is not one of the types specified by
-                TorchInputType or TorchTargetType
-
-        Returns:
-            Union[TorchTargetType, TorchInputType]: The data argument except now it's been moved to self.device
-        """
-        # Currently we expect both inputs and targets to be either tensors
-        # or dictionaries of tensors
-        if isinstance(data, torch.Tensor):
-            return data.to(self.device)
-        elif isinstance(data, dict):
-            return {key: value.to(self.device) for key, value in data.items()}
-        else:
-            raise TypeError(
-                "data must be of type torch.Tensor or Dict[str, torch.Tensor]. \
-                    If definition of TorchInputType or TorchTargetType has \
-                    changed this method might need to be updated or split into \
-                    two"
-            )
-
-    def is_empty_batch(self, input: Union[torch.Tensor, Dict[str, torch.Tensor]]) -> bool:
```

Comment:
This is a pretty generic function. Moved it to `utils/client.py`. Let me know if you disagree with the change.